spec/models/program_spec.rb

```diff
@@ -199,4 +199,21 @@
     end
   end
 
+  describe '#any_event_for_this_date?' do
+    it 'returns type should be boolean' do
+      expect(program.any_event_for_this_date?('').in?([true, false])).to eq(true)
+    end
+
+    it 'returns todays events if date is nil' do
+      date = DateTime.parse(' 00:00').utc
+      events = EventSchedule.where(schedule: program.selected_schedule).where(start_time: date..(date + 1))
+      expect(program.any_event_for_this_date?('')).to eq(events.any?)
+    end
+
+    it 'returns todays events if date is empty' do
+      date = DateTime.parse(' 00:00').utc
+      events = EventSchedule.where(schedule: program.selected_schedule).where(start_time: date)
+      expect(program.any_event_for_this_date?('')).to eq(events.any?)
+    end
+  end
 end
```

Comment:
You need to test what happens when 
* date is nil
* date is a valid date

and when 
* there is no selected_schedule
* when there are event schedule, but not event starts at desired time
* when there are event schedules, but there is a desired time